### PR TITLE
Make all handlers emit an `IbcEvent` with height` ctx.host_height()`

### PR DESCRIPTION
**Pull request by plafer**
Closes: #2035 

## Description

Make all handlers emit an `IbcEvent` with height `ctx.host_height()`.

______

### PR author checklist:
- [x] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [x] Added tests: integration (for Hermes) or unit/mock tests (for modules). 
- [x] Linked to GitHub issue.
- [x] Updated code comments and documentation (e.g., `docs/`).

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: modules/src/core/ics04_channel/handler/chan_close_init.rs [hunk 1]
```diff
@@ -70,3 +71,82 @@ pub(crate) fn process(
...
+        };
+
+        let handler_output =
+            channel_dispatch(&context, ChannelMsg::ChannelCloseInit(msg_chan_close_init)).unwrap();
+
+        assert!(!handler_output.events.is_empty()); // Some events must exist.
+
+        for event in handler_output.events.iter() {
+            assert!(matches!(event, &IbcEvent::CloseInitChannel(_)));
+            assert_eq!(event.height(), context.host_height());
```
Comment (adizere) [inline modules/src/core/ics04_channel/handler/chan_close_init.rs:149]: Cool, thanks for adding mock tests here and for channel close confirm!

File: modules/src/core/ics04_channel/handler/send_packet.rs [hunk 2]
```diff
@@ -59,9 +59,8 @@ pub fn send_packet(ctx: &dyn ChannelReader, packet: Packet) -> HandlerResult<Pac
 
     // check if packet height is newer than the height of the latest client state on the receiving chain
     let latest_height = client_state.latest_height();
-    let packet_height = packet.timeout_height;
 
-    if !packet.timeout_height.is_zero() && packet_height <= latest_height {
+    if !packet.timeout_height.is_zero() && packet.timeout_height <= latest_height {
```
Comment (adizere) [inline modules/src/core/ics04_channel/handler/send_packet.rs:63]: This change makes sense.

On a related topic, there is a comment above (L60) that is confusing me, specifically:

>    // check if packet height is newer than the height of the latest client state on the receiving chain

The receiving chain is the local chain (the one hosting & running the current ics04 module). It's not obvious to me what is the connection we meant to make between "latest client state" and "packet height". We should clarify that, or remove the comment if stale.

Comment (plafer) [reply modules/src/core/ics04_channel/handler/send_packet.rs:63]: It doesn't make sense to me either, so I'll just remove it.

Comment (plafer) [reply modules/src/core/ics04_channel/handler/send_packet.rs:62]: Removed a few comments that were either confusing or didn't say much [f8525a6867](https://github.com/informalsystems/ibc-rs/pull/2043/commits/f8525a6867c72f17e5d639363f2af0f6e6382fd9)

**Review by romac — APPROVED** [review]
Thanks @plafer 🙏

**Pull request merged**